freqtrade/rpc/rpc.py

```diff
@@ -963,6 +963,24 @@ def _convert_dataframe_to_dict(strategy: str, pair: str, timeframe: str, datafra
                 sell_mask = (dataframe['sell'] == 1)
                 sell_signals = int(sell_mask.sum())
                 dataframe.loc[sell_mask, '_sell_signal_close'] = dataframe.loc[sell_mask, 'close']
+            """
+            band-aid until this is fixed:
+            https://github.com/pandas-dev/pandas/issues/45836
+            """
+            datetime_types = ['datetime', 'datetime64', 'datetime64[ns, UTC]']
+            date_columns = dataframe.select_dtypes(include=datetime_types)
+            for date_column in date_columns:
+                # replace NaT with empty string,
+                # because if replaced with `None`
+                # it will be casted into NaT again
+                dataframe[date_column] = dataframe[date_column].apply(
+                    lambda x: '' if isnull(x) else x)
+
+            """
+            try this if above pandas Issue#45836 is fixed:
+            https://github.com/pandas-dev/pandas/issues/45836
+            """
+            # dataframe = dataframe.replace({NaT: None})
```

Comment:
I don't think this is a correct fix (or rather, a complete fix).

If you look at both my issue in pandas, as well as the one that's linked there, you'll notice that the same exact issue is also present for `NAN`.

The proper fix (fixing the issue on both date and numeric columns) is posted in the 2nd issue (https://github.com/pandas-dev/pandas/issues/45601#issuecomment-1028480734) - and would result in the below line to be changed to `<...>.astype(object).replace(...`.

currently, this is only fixing the issue for dates - but if a numeric column contains `inf` - pandas would first replace that wit NaN (This still works) - but would then fail to convert NAN to None (None is not a valid numeric numpy type).
Therefore the correct approach (as pointed out in the linked comment) is to first convert the column to object - and then eventually values to `None` (which is not a valid pandas numeric type).
I don't like it as it's broad, and will change the type of more columns than necessary - but the impact is limited as we're working on a copy of the dataframe anyway.

Reply:
I agree that this is a partial fix, I mentioned it in a comment further down. But you know the saying, perfect is the enemy of good and I learned a little bit more about ft webserver side.

But I don't mind if you don't want to merge this in yet, I just thought I'd share some temporary solutions if somebody needs it.

Talking about this:
> first convert the column to object - and then eventually values to None

Do you want me to implement this or should we wait for a better fix of pandas team?

Reply:
My point here is - if we implement a temporary fix - i'd want it to be complete - fixing both issues (even though no-one ran into the 2nd one yet - that's only a matter of time as the pandas-team confirmed that 
 this regression is present.

Reply:
Agree.

Tell me when or if I should do start working on this

Reply:
seems like the fix works - but i've simplified it a bit anyway and added a test :+1: